test_vm/tests/power_scenario_tests.rs

```diff
@@ -85,3 +99,87 @@ fn create_miner_test() {
     };
     expect.matches(v.take_invocations().last().unwrap())
 }
+
+#[test]
+fn test_cron_tick() {
+    let store = MemoryBlockstore::new();
+    let mut vm = VM::new_with_singletons(&store);
+
+    let addrs = create_accounts(&vm, 1, BigInt::from(10_000u64) * BigInt::from(10u64.pow(18)));
+
+    // create a miner
+    let (id_addr, robust_addr) = create_miner(
+        &mut vm,
+        addrs[0],
+        addrs[0],
+        RegisteredPoStProof::StackedDRGWindow32GiBV1,
+        TokenAmount::from(10_000e18 as i128),
+    );
+
+    // create precommit
+    let seal_proof = RegisteredSealProof::StackedDRG32GiBV1P1; // p1??
+    let sector_number = 100;
+    let sealed_cid = make_sealed_cid(b"100");
+    let precommit_params = PreCommitSectorParams {
+        seal_proof,
+        sector_number,
+        sealed_cid,
+        seal_rand_epoch: vm.get_epoch() - 1,
+        deal_ids: vec![],
+        expiration: vm.get_epoch()
+            + MIN_SECTOR_EXPIRATION
+            + max_prove_commit_duration(&Policy::default(), seal_proof).unwrap()
+            + 100,
+        ..Default::default()
+    };
+
+    apply_ok(
+        &vm,
+        addrs[0],
+        robust_addr,
+        TokenAmount::from(0u32),
+        MinerMethod::PreCommitSector as u64,
+        precommit_params,
+    );
+
+    // find epoch of miner's next cron task (precommit:1, enrollCron:2)
+    let cron_epoch = miner_dline_info(&vm, id_addr).last() - 1;
+
+    // create new vm at epoch 1 less than epoch requested by miner
+    let v = vm.with_epoch(cron_epoch);
+
+    // run cron and expect a call to miner and a call to update reward actor params
+    apply_ok(
+        &v,
+        *CRON_ACTOR_ADDR,
+        *STORAGE_POWER_ACTOR_ADDR,
+        BigInt::zero(),
+        PowerMethod::OnEpochTickEnd as u64,
+        RawBytes::default(),
+    );
+
+    let sub_invocs = vec![
+        // get data from reward and power for any eventual calls to confirmsectorproofsvalid
+        ExpectInvocation {
+            to: *REWARD_ACTOR_ADDR,
+            method: RewardMethod::ThisEpochReward as u64,
+            ..Default::default()
+        },
```

Comment:
Like I mentioned in the last round of review this is missing one of the key checks of the test: checking that the power actor cron calls back to the miner actor.   From the go test there are three expected subinvocations.  The two you have and then a third:
```golang
			{
				// expect call back to miner that was set up in create miner
				To:     minerAddrs.IDAddress,
				Method: builtin.MethodsMiner.OnDeferredCronEvent,
				From:   builtin.StoragePowerActorAddr,
				Value:  vm.ExpectAttoFil(big.Zero()),
			},
```
We need to put this expectation here to match the test behavior.

Reply:
I think part of the reason you went down this path is that I was off by one [here](https://github.com/filecoin-project/builtin-actors/pull/451#discussion_r892519292).  Change this expression to 
```rust
let cron_epoch = miner_dline_info(&v, miner_addrs.id_address).last()
```
 and then the test as written will fail with:
```
 expected: 
[
0: [f02:3],
1: [f02:4],
]
 was:
[
0: [f02:3],
1: [f0104:12],
2: [f02:4],
]

```
at which point all you need to do is add the expectation mentioned above for everything to pass

Reply:
(The `[f0104:12]` call is power actor --> miner)

Reply:
Ah that make's sense, pretty big overlook here but its resolved